awpy/visualization/plot.py

```diff
@@ -326,3 +346,369 @@ def plot_nades(
+    given_fig_ax: tuple[plt.Figure, plt.Axes] | tuple[None, None] = (None, None),
+    save_filepath: str = "",
+) -> None:
+    """Visualize map control for given ct/t map control value dictionaries.
+
+    Args:
+        map_name (str): Map used position_transform call
+        tile_values (FrameMapControlValues): Object contain map control values
+            for both sides
+        player_pos (FrameTeamMetadata): Object with metadata (player
+            positions, etc.) for each team.
+            Expected format same as output of extract_player_positions
+        given_fig_ax: Optional tuple containing figure and ax objects for plotting
+        save_filepath (str) : Filepath to save the figure to file if required
+
+    Returns: Nothing, all plotting is done on ax object
+
+    Raises:
+        ValueError: If map_name is not in awpy.data.NAV
+    """
+    if map_name not in NAV:
+        msg = "Map not found."
+        raise ValueError(msg)
+
+    figure, base_ax = given_fig_ax
+    if figure is None:
+        figure, base_ax = plot_map(map_name=map_name, map_type="simpleradar", dark=True)
+    _plot_map_control_snapshot_helper(
+        map_name, tile_values, base_ax, player_data=player_pos
+    )
+    if save_filepath:
+        figure.savefig(fname=save_filepath, bbox_inches="tight", dpi=400)
+        plt.close(figure)
+    elif not given_fig_ax[0]:
+        plt.show()
+        plt.close(figure)
+
+
+def plot_frame_map_control(
+    map_name: str,
+    frame: GameFrame,
+    plot_type: str = "",
+    given_fig_ax: tuple[plt.Figure, plt.Axes] | tuple[None, None] = (None, None),
+    save_filepath: str = "",
+) -> None:
+    """Visualize map control for awpy frame.
+
+    Args:
+        map_name (str): Map used position_transform call
+        frame (GameFrame): awpy frame to calculate map control for
+        plot_type (str): Determines which type of plot is created
+            (either with or without players)
+        given_fig_ax: Optional tuple containing figure and ax objects for plotting
+        save_filepath (str): Filepath to save the figure to file if required
+
+    Returns: Nothing, all plotting is done on ax object
+
+    Raises:
+        ValueError: If map_name is not in awpy.data.NAV
+    """
+    if map_name not in NAV:
+        msg = "Map not found."
+        raise ValueError(msg)
+
+    player_positions = extract_teams_metadata(frame)
+    map_control_dict = calc_parsed_frame_map_control_values(map_name, player_positions)
+    if plot_type == "players":
+        plot_map_control_snapshot(
+            map_name,
+            map_control_dict,
+            player_pos=player_positions,
+            given_fig_ax=given_fig_ax,
+            save_filepath=save_filepath,
+        )
+    else:
+        plot_map_control_snapshot(
+            map_name,
+            map_control_dict,
+            given_fig_ax=given_fig_ax,
+            save_filepath=save_filepath,
+        )
+
+
+def create_round_map_control_gif(
```

Comment:
One way to test could be to test whether the gif/images exist after they are created.